span.go

```diff
@@ -47,6 +35,46 @@ type Span interface {
 	// Suitable for serializing over the wire, etc.
 	TraceContext() TraceContext
 
+	// Log() records `data` to this Span at `timestamp`.
+	//
+	// See LogData for semantic details.
+	Log(timestamp time.Time, data LogData)
+
+	// Info() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprint(args...)})
+	//
+	Info(args ...interface{})
+
+	// Infof() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprintf(format, args...)})
+	//
+	Infof(format string, args ...interface{})
+
+	// Error() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprint(args...), IsError: true})
+	//
+	Error(args ...interface{})
+
+	// Errorf() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprintf(format, args...), IsError: true})
+	//
+	Errorf(format string, args ...interface{})
+
+	// RecordEvent() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Event: event})
+	//
+	// if payload is unspecified; otherwise, only one payload argument is
+	// accepted, and RecordEvent() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Event: event, Payload: payload[0]})
+	//
+	RecordEvent(event string, payload ...interface{})
```

Comment:
@yurishkuro +1 -- Had the same thought, based on this naming was wondering if `Event` is reserved as some sort of data structure